src/viewport/viewport.service.ts

```diff
@@ -100,72 +52,50 @@ export class ViewportService {
 
 		this.sizeType$ = this.resize$.pipe(
 			startWith(this.getViewportSize()),
-			map(x => this.calculateViewportSize(x.width)),
+			distinctUntilChanged((a, b) => a.width === b.width),
+			map(x => this.getWidthSizeInfo(x.width)),
```

Comment:
adjusted to retrieve the width size info on width change.

`calculateViewportSize` has been rewritten to retrieve only the viewport size